src/request.ts

```diff
@@ -37,23 +36,57 @@ export async function request<Query, Variables>(
 
     const url = ctx.apiUrl || defaultUrl;
 
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const { data: res, headers: responseHeaders } = await axios.post(
-      url,
-      {
+    const result = await fetch(url, {
+      method: 'POST',
+      headers,
+      body: JSON.stringify({
         query: query,
         variables: args.variables || null,
-      },
-      {
-        headers,
-      }
-    );
+      }),
+    });
 
-    if (!isPlainSuccessfulGraphQLResponse(res)) {
+    const status = result.status;
+    const responseHeaders = result.headers;
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const response = await result.json();
+
+    if (!isPlainSuccessfulGraphQLResponse(response)) {
       throw new Error('Unexpected response received');
     }
 
-    const mutationError = getMutationErrorFromResponse(res.data);
+    if (status === 401 || status === 403) {
```

Comment:
Technically this and the 500 branch can be done before `result.json()

Reply:
yep but it doesn't make any sense to do so for us since we want to use the response in those cases. Specifically currently when you get a 401 you are not told which permissions are actually missing which is super annoying. I'd like to be able to use the response body in that error case specifically.